spec/services/newspaper_works/text_extraction_derivative_service_spec.rb

```diff
@@ -51,5 +58,25 @@ def derivative_exists(ext)
       loaded_result = JSON.parse(File.read(json_path))
       expect(loaded_result['words'].length).to be > 1
     end
+
+    it "usually uses OCR, when no existing text" do
+      service = described_class.new(valid_file_set)
+      # here, service will delegate create_derivatives to OCR impl method:
+      expect(service).to receive(:create_derivatives_from_ocr)
```

Comment:
I find the the order of these two lines (both in this spec and the one below it) a bit confusing.
```
expect(service).to receive(:create_derivatives_from_ocr)
service.create_derivatives(source_image('ocr_mono.tiff'))
```
Is the call to `create_derivatives` required for the `expect` to pass? If so, should it be before?

I may be off-track here, sorry!

Reply:
All the examples I've seen with the `receive` matcher have the expectation marked before the call.  This seems counter-intuitive, but I think it is required by RSpec's implementation, or at least I get that sense from RSpec docs "Expecting messages" section.

I think this is because the matcher sets up some kind of spy, but do not know for sure.  I previously thought this was only convention for doubles, but in this case, the `create_derivatives` method will really call `create_derivatives_from_ocr` (in this example), and RSpec will notice that it was called (instrumented somehow beyond my understanding at the moment).